WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -23,37 +23,45 @@ public LabelSelectionViewModel(Money targetAmount)
 		_targetAmount = targetAmount;
 	}
 
-	public Pocket[] AllPockets { get; set; } = Array.Empty<Pocket>();
+	public Pocket[] NonPrivatePockets { get; set; } = Array.Empty<Pocket>();
 
 	public IEnumerable<LabelViewModel> AllLabelsViewModel { get; set; } = Array.Empty<LabelViewModel>();
 
 	public IEnumerable<LabelViewModel> LabelsWhiteList => AllLabelsViewModel.Where(x => !x.IsBlackListed);
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] GetSafeToUsePockets()
+	public Pocket[] AutoSelectPockets()
 	{
-		var pockets = AllPockets.ToList();
+		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
+		var allPockets = NonPrivatePockets.ToList();
 
 		if (_privatePocket is { } privatePocket)
 		{
-			pockets.Add(privatePocket);
+			allPockets.Add(privatePocket);
 		}
 
-		var unlabelledPocket = pockets.FirstOrDefault(x => x.Labels == CoinPocketHelper.UnlabelledFundsText);
+		if (_privatePocket is { } && _privatePocket.Amount >= _targetAmount)
```

Comment:
make `_privatePocket` non-nullable so you don't have to check for nulls all the time. Then move things where they make more sense, for example, don't modify the `allPockets` if you don't need. Please make the code simpler and cleaner.

Reply:
Done, but I have seen some more opportunities for simplification which should be done in a separate PR because they are out of the scope of this PR.